File: 01_Basico/03 - Lacos_Repeticao/Respsota_em_Python/questao12.py
produto_num = 1
total = 0

while True:
    print("**" * 20)
    print("Para encerrar essa compra digite 0")
    produto = float(input(f"Digite o valor do produto {produto_num}: "))
    produto_num += 1
    total += produto
    if produto == 0:
        print(f"Sua conta é: {total}")
        dinheiro = float(input("Insira o valor em dinheiro: "))
        troco = dinheiro - total
        print("**" * 20)
        print("Nota Fiscal")
        print(
            f"Total:            R${total}\n"
            f"Dinheiro:         R${dinheiro}\n"
            f"Troco:            R${troco}")
        encerrar = input("Deseja fechar o caixa ? S/N: ")
        produto_num = 1
        total = 0
        if encerrar[0].lower() == "s":
            break

File: 01_Basico/03 - Lacos_Repeticao/Resposta_em_Java_Script/questao12.js
let valor = 0;

while (true) {
    console.log("*".repeat(40));
    console.log("Para encerrar a compra digite 0");
    console.log("*".repeat(40));
    let produto = parseInt(prompt(`Digite o valor do produto ${valor + 1}: `));
    valor += produto;

    if (produto == 0) {
        console.log(`O valor total é:       R$${valor}`);
        let pagamento = parseFloat(prompt("Digite o valor que deseja pagar: "));
        let troco = pagamento - valor;
        console.log("*".repeat(40));
        console.log("Nota fiscal");
        console.log(`Total da Compra:       R$${valor}`);
        console.log(`Valor do pagamento:    R$${pagamento}`);
        console.log(`Troco:                 R$${troco}`);

        let fecharCaixa = prompt("Deseja fechar o caixa? S/N: ");
        if (fecharCaixa[0].toUpperCase() == "S") {
            break;
        }
        valor = 0;
    }
}
